Sources/SegmentAmplitude/AmplitudeSession.swift

```diff
@@ -197,15 +225,31 @@ extension AmplitudeSession {
     }
     
     private func startNewSession() {
+        if resetPending { return }
+        resetPending = true
+        sessionID = newTimestamp()
+        if eventSessionID == -1 {
+            // we only wanna do this if we had nothing before, so each
+            // event actually HAS a sessionID of some kind associated.
+            eventSessionID = sessionID
+        }
         analytics?.track(name: Constants.ampSessionStartEvent)
     }
     
     private func startNewSessionIfNecessary() {
+        if eventSessionID == -1 {
+            // we only wanna do this if we had nothing before, so each
+            // event actually HAS a sessionID of some kind associated.
+            eventSessionID = sessionID
+        }
```

Comment:
Should this be a function? Second time we see this code, and that `sessionID == -1` test is used a bunch too.

Reply:
sessionID == -1 is only used once.  eventSessionID == -1 is used twice.  it could be, but doesn't seem critical as I hope to never touch this again. :D